### PR TITLE
Bring in fixes for geometry modules

### LINKED ISSUE #1204: Geometry util bugfixes: frame_transform with multiple position vectors, cartesian_to_spherical units
### Requirements:

Two small changes, originally made in the much-too-big PR #1202 :

1. `geometry.frame_transform` must be slightly modified to allow multiple position vectors for a single event time
2. `geometry.cartesian_to_spherical` needed its documentation fixed (had incorrectly specified its return units). Modified it so that it has the same default behavior of returning degrees, but can return radians if needed. 

### Related Issues/PRs:
#1202

### PR DESCRIPTION
**Pull request by nkerman**
Closes #1204 

Two small changes, originally made in the much-too-big PR https://github.com/IMAP-Science-Operations-Center/imap_processing/pull/1202 :

1. geometry.frame_transform must be slightly modified to allow multiple position vectors for a single event time
2. geometry.cartesian_to_spherical needed its documentation fixed (had incorrectly specified its return units). Modified it so that it has the same default behavior of returning degrees, but can return radians if needed.

# Change Summary

## Updated Files

- imap_processing/spice/geometry.py
   - update frame_transform and cartesian_to_spherical functions

### CODE REVIEW

**Review by tmplummer — COMMENTED** [review]
All the changes look great. Can you add test coverage for the newly allowed input of a single `et` with multiple `position` vectors?

**Comment by nkerman** [comment]
> All the changes look great. Can you add test coverage for the newly allowed input of a single `et` with multiple `position` vectors?

Done! Thanks!

File: imap_processing/spice/geometry.py [hunk 1]
```diff
@@ -512,8 +521,12 @@ def cartesian_to_spherical(
         the spherical coordinates (r, azimuth, elevation):
 
         - r : Distance of the point from the origin.
-        - azimuth : angle in the xy-plane in radians [0, 2*pi].
-        - elevation : angle from the z-axis in radians [-pi/2, pi/2].
+        - azimuth : angle in the xy-plane
+          In degrees by default, if degrees parameter is True [0, 360],
+          otherwise in radians if degrees==False [0, 2*pi].
+        - elevation : angle from the z-axis
+          In degrees by default, if degrees parameter is True [0, 360],
```
Comment (laspsandoval) [inline imap_processing/spice/geometry.py]: I think there is a typo here?

**Review by laspsandoval — APPROVED** [review]
Looks good to me. Nice work! Just a single comment.

File: imap_processing/spice/geometry.py (hunk 1 above)
Comment (nkerman) [reply imap_processing/spice/geometry.py]: Ah, yep, thanks! it should be [0,180] for elevation. I tried to make the whole docstring clearer

**Pull request merged**